### PR TITLE
Calendar UI

### PR DESCRIPTION
**Pull request by pldespaigne**
solve some items of #7358

### CODE REVIEW

File: libs/contract/src/lib/avails/new-avails.ts [hunk 1]
```diff
@@ -379,7 +380,20 @@ function getMatchingCalendarSales<T extends (FullSale | BucketContract)>(sales:
   }));
 }
 
-function durationAvailabilities(
+function isCalendarTermInBucket<T extends BucketTerm | Term>(term: T, avails: CalendarAvailsFilter) {
```
Comment (RemcoSimonides) [inline libs/contract/src/lib/avails/new-avails.ts:383]: Why not call `isCalenderTermInAvails` directly? apperently being in avails and in bucket is the same thing?

File: libs/contract/src/lib/avails/new-avails.ts [hunk 2]
```diff
@@ -379,7 +380,20 @@ function getMatchingCalendarSales<T extends (FullSale | BucketContract)>(sales:
...
+  return isCalendarTermInAvails(term, avails);
+}
+
+function isCalendarTermSelected<T extends BucketTerm | Term>(term: T, avails: CalendarAvailsFilter) {
+  const exclusivityCheck =avails.exclusive === term.exclusive;
```
Comment (RemcoSimonides) [inline libs/contract/src/lib/avails/new-avails.ts:388]: missing space after `=`

File: libs/contract/src/lib/avails/new-avails.ts [hunk 3]
```diff
@@ -402,14 +416,23 @@ function durationAvailabilities(
...
-  return { available, sold, inBucket };
+  const inBucket: DurationMarker[] = [];
+  let selected: DurationMarker = undefined;
+
+  for (const bucketSale of bucketContracts ?? []) {
+    for (const term of bucketSale.terms) {
+      const isInBucket = isCalendarTermInBucket(term, avails);
+      const isSelected = isCalendarTermSelected(term, avails);
+      if (isSelected) {
+        selected = { from: term.duration.from, to: term.duration.to }; // , term, contract: bucketSale });
```
Comment (RemcoSimonides) [inline libs/contract/src/lib/avails/new-avails.ts:428]: I don't understand the comment behind this line. Is it some lost code?

**Review by RemcoSimonides — APPROVED** [review]

File: libs/contract/src/lib/avails/new-avails.ts (hunk 1 above)
Comment (pldespaigne) [reply libs/contract/src/lib/avails/new-avails.ts:383]: I could, it's just to make more readable and more clear about what is expected on line 425

**Pull request merged**